packages/nimble-components/src/table-column/number-text/index.ts

```diff
@@ -50,9 +54,14 @@ export class TableColumnNumberText extends TableColumnTextBase {
         this.updateColumnConfig();
     }
 
+    private alignmentChanged(): void {
+        this.updateColumnConfig();
+    }
+
     private updateColumnConfig(): void {
         const columnConfig: TableColumnNumberTextColumnConfig = {
-            formatter: this.createFormatter()
+            formatter: this.createFormatter(),
+            rightAlign: this.shouldRightAlign()
```

Comment:
Can we pass through as `alignment` instead? We are effectively resolving from a numericcalignment to a base alignment, so could just pass through that resolved value since that's what's used  (instead of coming up with a third intermediate)